app/routes/__index/certificate/index.tsx

```diff
@@ -85,13 +85,13 @@ export default function CertificateIndexRoute() {
     return date;
   }
 
-  if (certificate?.status === 'pending') {
-    return (
-      <Loading
-        img={pendingSvg}
-        desc="We have received your request, and will notify you when your certificate is ready"
-      />
-    );
+  function checkIfRenewValid(): boolean {
```

Comment:
ohh that's a good idea, I'll keep useMemo in mind with re-rendering a calculation multiple times

Reply:
BTW... should this also be using `useMemo`?

```
  function formatDate(val: Date): string {
    let date = val.toLocaleDateString('en-US', {
      month: 'short',
      day: '2-digit',
      year: 'numeric',
    });

    return date;
  }
  ```
  Just want to make sure I'm understanding its purpose...

Reply:
I think there's no need to do that. The function is called to format date when a value is passed in (line 127-128), while `useMemo` hook triggers when a dependency is updated.  From what I read, it's to [**memo**ized a value](https://www.w3schools.com/react/react_usememo.asp) to avoid having to recalculate it.

However, I notice there's two copies of the same function `formatDate`? On line 68-76 and 90-98.

Reply:
I think the second `formatDate` on line 90-98 is not needed?

Reply:
That's odd how did it get duplicated